components/style/properties/helpers.mako.rs

```diff
@@ -315,115 +315,106 @@
-                                    // In the case of a vector property we want to pass down
-                                    // an iterator so that this can be computed without allocation
-                                    //
-                                    // However, computing requires a context, but the style struct
-                                    // being mutated is on the context. We temporarily remove it,
-                                    // mutate it, and then put it back. Vector longhands cannot
-                                    // touch their own style struct whilst computing, else this will panic.
-                                    let mut s = context.mutate_style().take_${data.current_style_struct.name_lower}();
-                                    {
-                                        let iter = specified_value.compute_iter(context);
-                                        s.set_${property.ident}(iter ${maybe_cacheable});
-                                    }
-                                    context.mutate_style().put_${data.current_style_struct.name_lower}(s);
-                                % else:
-                                    let computed = specified_value.to_computed_value(context);
-                                     % if property.ident == "font_size":
-                                         longhands::font_size::cascade_specified_font_size(context,
-                                                                                           specified_value,
-                                                                                           computed,
-                                                                                           inherited_style.get_font());
-                                    % else:
-                                        context.mutate_style().mutate_${data.current_style_struct.name_lower}()
-                                               .set_${property.ident}(computed ${maybe_device}
-                                                                      ${maybe_cacheable} ${maybe_wm});
-                                    % endif
-                                % endif
+                if let Some(ref mut cascade_info) = *cascade_info {
+                    cascade_info.on_cascade_property(&declaration,
+                                                     &value);
+                }
+                % if property.logical:
+                    let wm = context.style.writing_mode;
+                % endif
+                <%
+                    maybe_wm = ", wm" if property.logical else ""
+                    maybe_cacheable = ", cacheable" if property.has_uncacheable_values == "True" else ""
+                    props_need_device = "content list_style_type".split() if product == "gecko" else []
+                    maybe_device = ", context.device" if property.ident in props_need_device else ""
```

Comment:
Yeah, I know this is just moving code around... It's just that seeing this makes me sad.